tests/Remove-DbaLinkedServer.Tests.ps1

```diff
@@ -17,56 +17,85 @@ Describe "$commandname Integration Tests" -Tags "IntegrationTests" {
+            $instance2.LinkedServers[$linkedServerName2].Drop()
         }
 
-        if ($instance2.LinkedServers.Name -contains "LS3_$random") {
-            $instance2.LinkedServers["LS3_$random"].Drop()
+        if ($instance2.LinkedServers.Name -contains $linkedServerName3) {
+            $instance2.LinkedServers[$linkedServerName3].Drop()
         }
+
+        if ($instance2.LinkedServers.Name -contains $linkedServerName4) {
+            $instance2.LinkedServers[$linkedServerName4].Drop($true)
+        }
+
+        Remove-DbaLogin -SqlInstance $instance2, $instance3 -Login $loginName -Confirm:$false
     }
 
     Context "ensure command works" {
 
         It "Removes a linked server" {
-            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer "LS1_$random"
+            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer $linkedServerName1
             $results.Length | Should -Be 1
-            Remove-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer "LS1_$random" -Confirm:$false
-            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer "LS1_$random"
+            Remove-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer $linkedServerName1 -Confirm:$false
+            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer $linkedServerName1
             $results | Should -BeNullOrEmpty
         }
 
         It "Tries to remove a non-existent linked server" {
-            Remove-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer "LS1_$random" -Confirm:$false -WarningVariable warnings
-            $warnings | Should -BeLike "*Linked server LS1_$random does not exist on $($instance2.Name)"
+            Remove-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer $linkedServerName1 -Confirm:$false -WarningVariable warnings
+            $warnings | Should -BeLike "*Linked server $linkedServerName1 does not exist on $($instance2.Name)"
         }
 
         It "Removes a linked server using a server from a pipeline and a linked server from a pipeline" {
-            $results = Get-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer "LS2_$random"
+            $results = Get-DbaLinkedServer -SqlInstance $script:instance2 -LinkedServer $linkedServerName2
             $results.Length | Should -Be 1
-            Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer "LS2_$random" | Remove-DbaLinkedServer -WarningVariable warn -Confirm:$false
+            Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer $linkedServerName2 | Remove-DbaLinkedServer -WarningVariable warn -Confirm:$false
             $warn | Should -BeNullOrEmpty
-            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer "LS2_$random"
+            $results = Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer $linkedServerName2
             $results | Should -BeNullOrEmpty
```

Comment:
Seems a bit excessive to make so many calls in one It block. Is there anything wrong with just having it test the below line. Not sure why we need to perform this test twice either (piping the command or the object via a variable is the same thing).

```
Get-DbaLinkedServer -SqlInstance $instance2 -LinkedServer $linkedServerName2 | Remove-DbaLinkedServer -WarningVariable warn -Confirm:$false
```

Reply:
The goal is to test both of the input object types. I split this into two separate tests.